processor/src/networks/bitcoin.rs

```diff
@@ -573,55 +679,24 @@ impl Network for Bitcoin {
         let offset_repr_ref: &[u8] = offset_repr.as_ref();
         let kind = kinds[offset_repr_ref];
 
-        let mut data = if kind == OutputType::External {
-          (|| {
-            for output in &tx.output {
-              if output.script_pubkey.is_op_return() {
-                match output.script_pubkey.instructions_minimal().last() {
-                  Some(Ok(Instruction::PushBytes(data))) => return data.as_bytes().to_vec(),
-                  _ => continue,
-                }
-              }
-            }
-            vec![]
-          })()
-        } else {
-          vec![]
-        };
-        data.truncate(MAX_DATA_LEN.try_into().unwrap());
-
-        let presumed_origin = {
-          let spent_output = tx.input[0].previous_output;
-          let mut spent_tx = spent_output.txid.as_raw_hash().to_byte_array();
-          spent_tx.reverse();
-          let spent_output = {
-            let mut tx;
-            while {
-              tx = self.get_transaction(&spent_tx).await;
-              tx.is_err()
-            } {
-              log::error!("couldn't get transaction from bitcoin node: {tx:?}");
-              sleep(Duration::from_secs(5)).await;
-            }
-            tx.unwrap().output.swap_remove(usize::try_from(spent_output.vout).unwrap())
-          };
-          BAddress::from_script(&spent_output.script_pubkey, BNetwork::Bitcoin).ok().and_then(
-            |address| {
-              let address = Address(address);
-              let encoded: Result<Vec<u8>, _> = address.clone().try_into();
-              if encoded.is_ok() {
-                Some(address)
-              } else {
-                None
-              }
-            },
-          )
-        };
-
-        let output = Output { kind, presumed_origin, output, data };
+        let output = Output { kind, presumed_origin: None, output, data: vec![] };
         assert_eq!(output.tx_id(), tx.id());
         outputs.push(output);
       }
+
+      if outputs.is_empty() {
+        continue;
+      }
+
+      // populate the rest of the outputs
+      let presumed_origin = self.extract_origin(tx).await;
+      let data = self.extract_serai_data(tx).await;
+      for output in &mut outputs {
+        if output.kind == OutputType::External {
+          output.data = data.clone();
+        }
+        output.presumed_origin = presumed_origin.clone();
+      }
```

Comment:
Since this is Bitcoin, should we collapse all of these outputs?

I'm unsure why someone would want to do `n` identical actions in a single transaction. If we don't allow multiple segments of data (such as data defined as `vec![return_data, ...segwit_inputs]`) ,let's continue your cleanup and simplify this :)